test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/ParameterAliasFunctionalTests.cs

```diff
@@ -335,6 +334,494 @@ public void ParseFilter_AliasInFunction_BuiltIn()
                     aliasNodes["@p1"].ShouldBeSingleValuePropertyAccessQueryNode(HardCodedTestModel.GetPeopleSet().EntityType().FindProperty("Name"));
                 });
         }
+
+        [Fact]
+        public void ParseFilter_AliasInFilterPathSegment_AliasAsFirstSegment()
+        {
+            Action parse = () => ParseUriAndVerify(
+                new Uri("http://gobbledygook/$filter=@p1&@p1=true"),
+                (oDataPath, filterClause, orderByClause, selectExpandClause, aliasNodes) =>
+                {
+                });
+            parse.ShouldThrow<ODataException>().WithMessage(ODataErrorStrings.RequestUriProcessor_ResourceNotFound("$filter=@p1&@p1=true"));
+        }
+
+        [Fact]
+        public void ParseFilter_AliasInFilterPathSegment_FilterSegmentOnSingleton()
+        {
+            Action parse = () => ParseUriAndVerify(
+                new Uri("http://gobbledygook/Boss/$filter=@p1?@p1=true"),

```

Comment:
>$filter=@p1?@p1=true [](start = 50, length = 20)

Would you please make sure $filter can follow up the singleton? What's the scenario?

Reply:
What is the scenario for $filter following singleton? I think it's correct that $filter cannot follow singleton because there is only one entity and filtering should apply on a collection.